code/modules/mob/living/basic/space_fauna/spider/spider_abilities/web.dm

```diff
@@ -83,3 +83,60 @@
 
 /datum/action/cooldown/lay_web/sealer/obstructed_by_other_web()
 	return !!(locate(/obj/structure/spider/stickyweb/sealed) in get_turf(owner))
+
+/datum/action/cooldown/lay_web/solid_web
+	name = "Spin Solid Web"
+	desc = "Spin a web to slow down potential prey."
+	button_icon_state = "lay_solid_web"
+	cooldown_time = 0 SECONDS
+	webbing_time = 5 SECONDS
+
+/datum/action/cooldown/lay_web/solid_web/obstructed_by_other_web()
+	return !!(locate(/obj/structure/spider/solid) in get_turf(owner))
+
+/datum/action/cooldown/lay_web/solid_web/plant_web(turf/target_turf, obj/structure/spider/stickyweb/existing_web)
+	new /obj/structure/spider/solid(target_turf)
+
+/datum/action/cooldown/lay_web/web_passage
+	name = "Spin Web Passage"
+	desc = "Spin a web passage to hide the nest from prey view."
+	button_icon_state = "lay_web_passage"
+	cooldown_time = 0 SECONDS
+	webbing_time = 4 SECONDS
+
+/datum/action/cooldown/lay_web/web_passage/obstructed_by_other_web()
+	return !!(locate(/obj/structure/spider/passage) in get_turf(owner))
```

Comment:
Should this be able to be placed on top of other solid or non-solid webs? Currently you are only preventing a passage from being overlaid onto another passage.
This is also true for all the following subtypes, so I'm not going to repeat it two more times.

It's starting to look like these structures should all share a subtype for ease of use.

Reply:
> Should this be able to be placed on top of other solid or non-solid webs? Currently you are only preventing a passage from being overlaid onto another passage.
> 
> This is also true for all the following subtypes, so I'm not going to repeat it two more times.
> 
> 
> 
> It's starting to look like these structures should all share a subtype for ease of use.

Passage is specifically designed to have spider spikes, normal webs and sticky webs below it putting a web passage and a solid web together is essentially just reinforcing a web which for the current change in going for with spiders being more of s defend the castle kind of antag im all about giving them the freedom to do so it also looks nice